clients/lsp-sqls.el

```diff
@@ -52,23 +52,124 @@
-                                ((equal lsp-sqls-workspace-config-path "workspace")
-                                 ".sqls/config.json")
-                                ((equal lsp-sqls-workspace-config-path "root")
-                                 (-> (lsp-workspace-root)
-                                     (f-join ".sqls/config.json"))))))
-    (when (file-exists-p config-json-path)
-      (lsp--set-configuration (lsp--read-json-file config-json-path)))))
+
+  (if lsp-sqls-connections
+      (lsp--set-configuration `(:sqls (:connections ,(apply #'vector lsp-sqls-connections))))
+    (when-let ((config-json-path (cond
+                                  ((equal lsp-sqls-workspace-config-path "workspace")
+                                   ".sqls/config.json")
+                                  ((equal lsp-sqls-workspace-config-path "root")
+                                   (-> (lsp-workspace-root)
+                                       (f-join ".sqls/config.json"))))))
+      (when (file-exists-p config-json-path)
+        (lsp--set-configuration (lsp--read-json-file config-json-path))))))
+
+(defun lsp-sqls--show-results (result)
+  (with-current-buffer (get-buffer-create "*sqls results*")
+    (with-help-window (buffer-name)
+      (erase-buffer)
+      (insert result))))
+
+(defun lsp-sql-execute-query (&optional command)
+  "Execute selected region/whole buffer against current database."
+  (interactive)
+  (lsp-sqls--show-results
+   (lsp-request
+    "workspace/executeCommand"
+    (list :command "executeQuery"
+          :arguments (or
+                      (when command
+                        (lsp:command-arguments? command))
+                      (vector (lsp--buffer-uri)))
+          :timeout lsp-sqls-timeout
+          :range (list
+                  :start (lsp--point-to-position
+                          (if (use-region-p)
+                              (region-beginning)
+                            (point-min)))
+                  :end (lsp--point-to-position
+                        (if (use-region-p)
+                            (region-end)
+                          (point-max))))))))
+
+(defun lsp-sql-show-databases (&optional _command)
+  "Show databases."
+  (interactive)
+  (lsp-sqls--show-results
+   (lsp-request
+    "workspace/executeCommand"
+    (list :command "showDatabases" :timeout lsp-sqls-timeout))))
+
+(defun lsp-sql-show-schemas (&optional _command)
+  "Show schemas."
+  (interactive)
+  (lsp-sqls--show-results
+   (lsp-request
+    "workspace/executeCommand"
+    (list :command "showSchemas" :timeout lsp-sqls-timeout))))
+
+(defun lsp-sql-show-connections (&optional _command)
+  "Show connections"
+  (interactive)
+  (lsp-sqls--show-results
+   (lsp-request
+    "workspace/executeCommand"
+    (list :command "showConnections" :timeout lsp-sqls-timeout))))
+
+(defun lsp-sql-switch-database (&optional _command)
+  "Switch database."
+  (interactive)
+  (lsp--send-execute-command
```

Comment:
This will error with `Please open an issue in lsp-mode for implementing switchDatabase` if anything goes wrong, even though it is implemented. Perhaps we need another type of `lsp--send-execute-command` to be used outside of lenses and actions for such cases, e.g. `lsp--command-execute` which doesn't have that handler. Something like:
```emacs-lisp
(defun lsp--command-execute (command &optional args)
  (let ((params (if args
                    (list :command command :arguments args)
                  (list :command command))))
    (lsp-request "workspace/executeCommand" params)))

(defun lsp--send-execute-command (command &optional args)
  (condition-case-unless-debug err
      (lsp--command-execute command args)
    (error
     (lsp--error "Please open an issue in lsp-mode for implementing `%s'.\n\n%S"
                 command err))))
```

Reply:
I will include that in the PR. I will make one of the functions public to make it easier to be used by the extensions.

Reply:
Actually, better to have it in separate PR since we have to change the callers as well.

Reply:
I added the methods you suggested without touching anything else. In sqls it is common to get an error from the server(e. g. unable to connect/execute query, etc) and it would have been very confusing if we were showing that error. I am also thinking about removing that message - it seems like it brings more confusion and it seems like it is incorrect in general, since the server might error even in the lenses case.